pkg/controller/agent/fleet.go

```diff
@@ -265,6 +285,34 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
 	return token
 }
 
+func getKibanaBasePath(ctx context.Context, client k8s.Client, kibanaNSN types.NamespacedName) (string, error) {
+	var kb v1.Kibana
+	err := client.Get(ctx, kibanaNSN, &kb)
+	if err != nil {
+		return "", err
+	}
+
+	if kb.Spec.Config == nil {
+		return "", nil
+	}
+
+	// Get Kibana Config secret
+	kbSecretName := kibana.SecretName(kb)
+	var kbConfigsecret corev1.Secret
+	err = client.Get(ctx, types.NamespacedName{Namespace: kb.Namespace, Name: kbSecretName}, &kbConfigsecret)
+	if err != nil {
+		return "", err
+	}
```

Comment:
Not asking for a change, but wanting to start the discussion on whether we want to handle `not found` any differently here, or in the calling block?

Reply:
I am thinking, if the secret is not found as well we want to error out and reconcile again and there shouldn't be where the secret doesn't exist right ?

Reply:
I guess what I was asking is if it's preferable to do something like this in the calling func to minimize noise in the logs:
```diff
diff --git a/pkg/controller/agent/fleet.go b/pkg/controller/agent/fleet.go
index 63cee6cc4..4f61d7ef2 100644
--- a/pkg/controller/agent/fleet.go
+++ b/pkg/controller/agent/fleet.go
@@ -19,6 +19,7 @@ import (
        "go.elastic.co/apm/module/apmhttp/v2"
        "gopkg.in/yaml.v3"
        corev1 "k8s.io/api/core/v1"
+       apierrors "k8s.io/apimachinery/pkg/api/errors"
        "k8s.io/apimachinery/pkg/types"
        "k8s.io/client-go/tools/record"
        "sigs.k8s.io/controller-runtime/pkg/reconcile"
@@ -266,7 +267,10 @@ func maybeReconcileFleetEnrollment(params Params, result *reconciler.Results) En
        }

        kibanaBasePath, err := getKibanaBasePath(params.Context, params.Client, params.Agent.Spec.KibanaRef.WithDefaultNamespace(params.Agent.Namespace).NamespacedName())
-       if err != nil {
+       if err != nil && apierrors.IsNotFound(err) {
+               result.WithResult(reconcile.Result{Requeue: true})
+               return EnrollmentAPIKey{}
+       } else if err != nil {
                result.WithError(err)
                return EnrollmentAPIKey{}
        }
```

Reply:
aah I see what you mean, that makes sense.

Reply:
done